tests/b037.py

```diff
@@ -0,0 +1,33 @@
+
+class A:
+    def __init__(self) -> None:
+        return 1  # bad
+
+class B:
+    def __init__(self, x) -> None:
+        if x:
+            return  # ok
+        else:
+            return []  # bad
+
+    class BNested:
+        def __init__(self) -> None:
+            yield  # bad
+
+
+class C:
+    def func(self):
+        pass
+
+    def __init__(self, k="") -> None:
+        yield from []  # bad
+
+
+class D(C):
+    def __init__(self, k="") -> None:
+        super().__init__(k)
+        return None  # bad
```

Comment:
I can see people disliking this due to

```
Explicit is better than implicit
```
In the Zen of Python. 

Can we just check if a `None` is being explicitly returned and not error in that case since we allow `return` ...

Reply:
Yeah it's easy enough to do if you prefer, but my argument against allowing `return None` is that, while `return` is functionally equivalent to `return None` in python, the latter should only be used in cases where you _might_ return something (like a C function that returns either a valid pointer or `NULL`). Whereas just bare `return` is ok, when used as a early exit point from a function (like a C void function).

some quick examples:

```python
def get_thing() -> Thing | None:
    """return a thing if possible"""
    for thing in things:
        if condition:
            return thing
    return None  # this line could be omitted, but functions that have a non-empty return
                 # in some case should have non-empty return in all cases (pylint would 
                 # complain about this for example)
```

vs 

```python
class Example:
    def __init__(self, size, do_extra_stuff = False):
        self.x = ...

        if not do_extra_stuff:
            return  # early exit. Could do `return None` but that implies
                    # that we might return something somewhere else?

        # do some more stuff
```

let me know what you think, thanks!

Reply:
I agree that it's better to lint against `return None`.

Reply:
Since this is `__init__` only, I can live with it. I'll have code to 'fix' cause of it.